src/ensembl/src/content/app/entity-viewer/types/gene.ts

```diff
@@ -1,11 +1,26 @@
+import JSONValue from 'src/shared/types/JSON';
 import { Slice } from './slice';
 import { Transcript } from './transcript';
+import { ExternalLink } from './externalLink';
+import { Source } from './source';
 
 export type Gene = {
   type: 'Gene';
   id: string;
   symbol: string;
+  name?: string;
+  source?: Source;
   so_term: string; // is there a better name for it?
   slice: Slice;
   transcripts: Transcript[];
+  synonyms?: string[];
+  attributes?: string[];
+  function?: {
+    description: string;
+    source: Source;
+  };
+  xrefs?: ExternalLink[];
```

Comment:
If we are not using a dedicated backend-for-frontend that modifies the original payload, then an external reference, as returned from the api, is likely to have a shape similar to this:

```
{
  "accession_id": "R-HSA-5693532",
  "name": "R-HSA-5693532",
  "description": "DNA Double-Strand Break Repair",
  "source_uri": "some string",
  "source": {
    "name": "Reactome Pathway",
    "url": "https://reactome.org"
  }
}
```
It's not yet certain that this is the final shape; but it certainly won't contain the `links` field of type `Source`

Reply:
I guess we can update this once we have the right structure later on?